Source/loadsave.cpp

```diff
@@ -1103,60 +1103,60 @@ void LoadGame(bool firstflag)
 			LoadLighting(&file, &VisionList[i]);
 	}
 
-	for (int i = 0; i < MAXITEMS; i++)
-		itemactive[i] = file.nextLE<Sint8>();
-	for (int i = 0; i < MAXITEMS; i++)
-		itemavail[i] = file.nextLE<Sint8>();
+	for (int &itemId : itemactive)
+		itemId = file.nextLE<Sint8>();
+	for (int &itemId : itemavail)
+		itemId = file.nextLE<Sint8>();
 	for (int i = 0; i < numitems; i++)
 		LoadItem(&file, itemactive[i]);
-	for (int i = 0; i < 128; i++)
-		UniqueItemFlag[i] = file.nextBool8();
+	for (bool &UniqueItemFlag : UniqueItemFlags)
+		UniqueItemFlag = file.nextBool8();
 
 	for (int j = 0; j < MAXDUNY; j++) {
-		for (int i = 0; i < MAXDUNX; i++)
-			dLight[i][j] = file.nextLE<Sint8>();

```

Comment:
I'd leave 2d arrays alone as this is less readable

Reply:
Mkay, it better to keep it with the other comments about the same pattern.